File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        h_s, h_t = {}, {}
        for i in range(len(s)):
            h_s[s[i]] = 1 + h_s.get(s[i],0)
            h_t[t[i]] = 1 + h_t.get(t[i],0)
        
        for i in h_s.keys():
            if h_s[i] != h_t.get(i, 0):
                return False
        return True
        
        #print(h_s)
        # for i in s:
        #     anagram = False
        #     for j in range(len(t)):
        #         if i == t[j] and not anagram:
        #             anagram = True
        #             t = t[0:j] + t[j+1:]
        #             break
        #     if not anagram:
        #         return False
        # if len(t) > 0:
        #     return False
        # return True

File: 242-valid-anagram/valid-anagram.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> h_s = new HashMap<Character, Integer>();
        HashMap<Character, Integer> h_t = new HashMap<Character, Integer>();
        if(s.length() != t.length()) return false;
        for(int i = 0; i < s.length(); i++){
            if (h_s.get(s.charAt(i))!=null){
                h_s.put(s.charAt(i), h_s.get(s.charAt(i)) + 1);
            }else{
                h_s.put(s.charAt(i), 1);
            }
            if (h_t.get(t.charAt(i))!= null){
                h_t.put(t.charAt(i), h_t.get(t.charAt(i)) + 1);
            }else{
                h_t.put(t.charAt(i), 1);
            }
        }
        for(char k: h_s.keySet()){
            if(h_t.get(k) == null || !h_s.get(k).equals(h_t.get(k))){
                return false;
            }
        }
        return true;
    }
}